Functions/xASL_bids_BIDSifyASLJSON.m

```diff
@@ -173,13 +173,24 @@
 end
     
 %% 7. Check for Hadamard sequence
+
+% Hadamard Check
+if isfield(jsonOut,'TimeEncodedMatrix') && ~isempty(jsonOut,'TimeEncodedMatrix') || ... % Should be 4, 8 or 12
+    isfield(jsonOut,'HadamardMatrixType') % Natural or walsh
+    bHadamard = true; 
+else 
+    bHadamard = false; 
+end
+
+
 % FME sequence check
 if isfield(jsonIn,'SeriesDescription')
-	isHadamardFME = ~isempty(regexp(char(jsonIn.SeriesDescription),'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
+	bHadamardFME = ~isempty(regexp(char(jsonIn.SeriesDescription),'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
```

Comment:
Do we only check HadamardFME here (which can be with or without multiple echo times!) or specifically the Hadamard FME sequence with multiple echo times? Be specific in the variable name

Reply:
But the format of the SeriesDescription of encoded Hadamard is the same here and two booleans are more than enough for the use-case here. Let's stick to bHadamard & bHadamardFME.

Reply:
bHadamardFME -> bTimeEncodedFME

Reply:
solved 👍